Formula/gi-docgen.rb

```diff
@@ -0,0 +1,86 @@
+
+  # Source for latest version is not available on PyPI, so using GitHub tarball instead.
+  # Issue ref: https://github.com/leohemsted/smartypants.py/issues/8
+  resource "smartypants" do
+    url "https://github.com/leohemsted/smartypants.py/archive/refs/tags/v2.0.1.tar.gz"
+    sha256 "b98191911ff3b4144ef8ad53e776a2d0ad24bd508a905c6ce523597c40022773"
+  end
+
+  resource "Jinja2" do
+    url "https://files.pythonhosted.org/packages/89/e3/b36266381ae7a1310a653bb85f4f3658c462a69634fa9b2fef76252a50ed/Jinja2-3.1.1.tar.gz"
+    sha256 "640bed4bb501cbd17194b3cace1dc2126f5b619cf068a726b98192a0fde74ae9"
+  end
+
+  resource "Markdown" do
+    url "https://files.pythonhosted.org/packages/15/06/d60f21eda994b044cbd496892d4d4c5c708aa597fcaded7d421513cb219b/Markdown-3.3.6.tar.gz"
+    sha256 "76df8ae32294ec39dcf89340382882dfa12975f87f45c3ed1ecdb1e8cefc7006"
+  end
+
+  resource "MarkupSafe" do
+    url "https://files.pythonhosted.org/packages/1d/97/2288fe498044284f39ab8950703e88abbac2abbdf65524d576157af70556/MarkupSafe-2.1.1.tar.gz"
+    sha256 "7f91197cc9e48f989d12e4e6fbc46495c446636dfc81b9ccf50bb0ec74b91d4b"
+  end
+
+  resource "Pygments" do
+    url "https://files.pythonhosted.org/packages/94/9c/cb656d06950268155f46d4f6ce25d7ffc51a0da47eadf1b164bbf23b718b/Pygments-2.11.2.tar.gz"
+    sha256 "4e426f72023d88d03b2fa258de560726ce890ff3b630f88c21cbb8b2503b8c6a"
+  end
+
+  resource "toml" do
+    url "https://files.pythonhosted.org/packages/be/ba/1f744cdc819428fc6b5084ec34d9b30660f6f9daaf70eead706e3203ec3c/toml-0.10.2.tar.gz"
+    sha256 "b3bda1d108d5dd99f4a20d24d9c348e91c4db7ab1b749200bded2f839ccbe68f"
+  end
+
+  resource "typogrify" do
+    url "https://files.pythonhosted.org/packages/8a/bf/64959d6187d42472acb846bcf462347c9124952c05bd57e5769d5f28f9a6/typogrify-2.0.7.tar.gz"
+    sha256 "8be4668cda434163ce229d87ca273a11922cb1614cb359970b7dc96eed13cb38"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    (testpath/"brew.toml").write <<~EOS
+      [library]
+      description = "Homebrew gi-docgen formula test"
+      authors = "Homebrew"
+      license = "BSD-2-Clause"
+      browse_url = "https://github.com/Homebrew/brew"
+      repository_url = "https://github.com/Homebrew/brew.git"
+      website_url = "https://brew.sh/"
+    EOS
+
+    (testpath/"brew.gir").write <<~EOS
+      <?xml version="1.0"?>
+      <repository version="1.2"
+                  xmlns="http://www.gtk.org/introspection/core/1.0"
+                  xmlns:c="http://www.gtk.org/introspection/c/1.0">
+        <namespace name="brew" version="1.0" c:identifier-prefixes="brew" c:symbol-prefixes="brew">
+          <record name="Formula" c:type="Formula">
+            <field name="name" writable="1">
+              <type name="utf8" c:type="char*"/>
+            </field>
+          </record>
+        </namespace>
+      </repository>
+    EOS
+
+    assert_equal "", shell_output("#{bin}/gi-docgen generate --quiet -C brew.toml brew.gir").chomp
```

Comment:
I think the issue is that upstream is writing to stdout across multiple threads and the interleaving is causing random corrupted data on Linux.

For now, using `--quiet` flag to avoid stdout. This may be required when using in dependents.

Reply:
Opened https://gitlab.gnome.org/GNOME/gi-docgen/-/issues/129